flink-clients/src/main/java/org/apache/flink/client/cli/ProgramOptions.java

```diff
@@ -119,6 +124,14 @@ else if (args.length > 0) {
 		} else {
 			this.savepointSettings = SavepointRestoreSettings.none();
 		}
+
+		if (line.hasOption(JOB_ID_OPTION.getOpt())) {
+			String jobIdRaw = line.getOptionValue(JOB_ID_OPTION.getOpt());
+			UUID id = UUID.fromString(jobIdRaw);
+			this.jobID = JobID.fromHexString(id.toString().replace("-", ""));
+		} else {
+			this.jobID = JobID.generate();
```

Comment:
It seems like the default should be to leave this null. If it's nullable, you don't need `JobID.generate()` scattered throughout the code, and it allows for the ability to distinguish between "the user asked for this specific JobID" and "the user doesn't care what JobID we use".